src/components/MarketMySharesForm/index.js

```diff
@@ -87,7 +82,10 @@ class MarketMySharesForm extends Component {
   @autobind
   handleSellShare(shareId, shareAmount) {
     const shareIndex = this.props.marketShares.map(share => share.id).indexOf(shareId)
-    return this.props.sellShares(this.props.market, shareIndex, shareAmount).then(() => this.props.reset())
+    const shareBalance = new Decimal(this.props.marketShares[shareIndex].balance).div(1e18).toDP(2, 1)
+    const selectedSellAmount = new Decimal(shareAmount)
+    const sellAmount = shareBalance.sub(selectedSellAmount).lt(MIN_CONSIDER_VALUE) ? shareBalance : shareAmount
+    return this.props.sellShares(this.props.market, shareIndex, sellAmount).then(() => this.props.reset())
```

Comment:
can we use `async` + `await` here instead of promises?

Reply:
@andre-meyer it was using promises, i didn't change anytyhing. I will then :smiley:

Reply:
Thank you! 👍